src/main.js

```diff
@@ -124,7 +125,16 @@ const annotationsPrototype = {
 
   getAnnotations() {
     const activeLayer = this.activeLayer;
-    return this.overlays[activeLayer].model.getAll();
+    return this.overlays[activeLayer].model.annotations;
+  },
+
+  getSelection() {
+    const activeLayer = this.activeLayer;
+    return this.overlays[activeLayer].model.selection;
+  },
+
+  cleanSelection() {
+    this.dispatch({ type: 'SELECTION_RESET', annotations })
   },
 
   setAnnotations(annotations) {
```

Comment:
when does setLayerAnnotations get called? Seems a little wierd to have a separate handler for it outside of setAnnotations

Reply:
This was a needed catch, I'll need to clean this up. setAnnotations calls the dispatcher, but each layer has it's own dispatcher. Calling the dispatcher would affect the model and view. 

setLayerAnnotations directly sets the annotations in the model for a non-active layer, as only one dispatcher is active at a time. So we couldn't change the annotations of background layers (that are still in view) by calling setAnnotations. 

This is definitely poor in hindsight. We should ideally have just one dispatcher hub, that knows which layer is active and needs re-rendering, and can also update the models of non-active layers. And dispatcher callbacks should be the only allowed source to mutate state.